ldapauthenticator/ldapauthenticator.py

```diff
@@ -403,13 +406,26 @@ def get_connection(self, userdn, password):
             port=self.server_port,
             use_ssl=use_ssl,
         )
-        conn = ldap3.Connection(
-            server,
-            user=userdn,
-            password=password,
-            auto_bind=auto_bind,
-        )
-        return conn
+        try:
+            self.log.debug(f"Attempting to bind {userdn}")
+            conn = ldap3.Connection(
+                server,
+                user=userdn,
+                password=password,
+                auto_bind=auto_bind,
+            )
+        except LDAPBindError as e:
+            self.log.warning(
```

Comment:
This log message is promoted from debug to warning, meaning that every failed login should expect `len(bind_dn_template) + 1` warning messages. Is that intentional? Should this still be debug-level?

Reply:
Ah good point, we don't want this to log warnings non-stop, because we also attempt to bind and is expecting to fail doing so sometimes - such as when we have multiple DN templates for a username.

I've adjusted this now so that its always debug, and when get_connection has returned None, a warning (failed to bind normal user) or error log (failed to bind administrative user to do searches) is emitted depending on the situation.

Reply:
Meant to be resolved with 1ba6e0f, I also fixed grammar and the deprecated use of `log.warn` (should be `log.warning`)